sfge/src/test/java/com/salesforce/rules/multiplemassschemalookup/MultipleMassSchemaLookupRuleTest.java

```diff
@@ -341,4 +367,46 @@ public void testLoopBeforeAndAroundGgd(String loopAstLabel, String loopStructure
                                 MmslrUtil.RepetitionType.LOOP)
                         .withOccurrence(loopAstLabel, MY_CLASS, 7));
     }
+
+
+
+    @CsvSource({
+        "ForEachStatement, for (String s : myList), ForEachStatement, for (String s : myList)",
+        "ForEachStatement, for (String s : myList), ForLoopStatement, for (Integer i; i < s.size; i++)",
+        "ForEachStatement, for (String s : myList), WhileLoopStatement, while(true)",
+        "ForLoopStatement, for (Integer i; i < s.size; i++), ForLoopStatement, for (Integer i; i < s.size; i++)",
+        "ForLoopStatement, for (Integer i; i < s.size; i++), ForEachStatement, for (String s : myList)",
+        "ForLoopStatement, for (Integer i; i < s.size(); i++), WhileLoopStatement, while(true)",
+        "WhileLoopStatement, while(true), ForEachStatement, for (String s : myList)",
+        "WhileLoopStatement, while(true), ForLoopStatement, for (Integer i; i < s.size; i++)",
+        "WhileLoopStatement, while(true), WhileLoopStatement, while(true)"
+    })
+    @ParameterizedTest(name = "{displayName}: {0}")
+    public void testNestedLoop(String outerLoopLabel, String outerLoopStructure, String innerLoopLabel, String innerLoopStructure) {
```

Comment:
Yes, you're right. Though it's a good thing I didn't see `testLoopWithinLoop` because this is how I caught the nested loop bug! I'll combine them.

Reply:
Actually, since adding these cases into `testLoopWithinLoop` would require refactoring that test to basically look like this one, could merge `testLoopWithinLoop` into this one?

Reply:
We can discuss this one more later.